---
title: "BC project.step1"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo                  = TRUE
                      ,warning              = FALSE
                      ,message              = FALSE
                      ,error                = TRUE
                      ,dev                  = c('png','pdf')
                      ,pdf.options(encoding = "ISOLatin9.enc", useKerning=FALSE, useDingbats=FALSE)
)                      
```

```{r libraries}
library(Seurat)
```

```{r reading the scRNA-seq data}
# Path to the count matrix of scRNA-seq data from BC patients 
path= "/mnt/data1/elyas/Single-cell/test-data/Breast-cancer4/"

BC_list <- list("CID3586","CID3838","CID3921","CID3941","CID3946","CID3948","CID3963","CID4040","CID4066","CID4067",
                "CID4290A","CID4398","CID44041","CID4461","CID4463","CID4465","CID4471","CID4495","CID44971","CID44991",
                 "CID4513","CID4515","CID45171","CID4523","CID4530N","CID4535")

# Reading in the count matrix of scRNA-seq data from BC patients 
BC.Se <- list()
BC <- for (i in BC_list) {
  BC.Se <- c(BC.Se,Read10X(paste0(path, i), gene.column=1))
}

# Creating Seurat objects from the 10x scRNA-seq profiles

BC.Se <-mapply(function(X,Y){
  CreateSeuratObject(X, project = Y,assay = "RNA")},X=BC.Se,Y=BC_list)

#Merging the objects
BC.Se <- Reduce(merge,BC.Se)

```

```{r adding the metadata}
# Reading in the metadata
BC_meta <- read.table("/mnt/data1/elyas/Single-cell/test-data/Breast-cancer4/metadata.csv", sep = ",", row.names = 1, header = T)

BC_meta <- BC_meta[match(rownames(BC.Se@meta.data),rownames(BC_meta)),]

BC.Se@meta.data <- BC_meta

```

```{r inferCNV}
# Extracting the data from patients whom their epithelial cells could be distinguished to malignant and normal status.
BC.Se.mal <- SplitObject(BC.Se, split.by = "normal_cell_call")
BC.Se.mal <- BC.Se.mal$cancer

knitr::kable(table(BC.Se.mal$Patient,BC.Se.mal$celltype_major),caption = "The patients whom their epithelial cells \ncould be distinguished to malignant and normal status")


BC.Se <- SplitObject(BC.Se, split.by = "Patient")

# Choosing only the profiles from patients with distinguished malignant epithelial cells 
BC.Se.inferCNV <- list()
for (i in c(unique(BC.Se.mal$Patient))) {
 
 BC.Se.inferCNV <- c(BC.Se.inferCNV, BC.Se[[i]])
} 

BC.Se.inferCNV <- Reduce(merge, BC.Se.inferCNV)

```

```{r CD4}
# Extracting the CD4+ T cells
CD4 <- SplitObject(BC.Se.inferCNV, split.by = "celltype_minor")
CD4 <- CD4$`T cells CD4+`
```


```{r Dimensionality reduction}
# Dimensionality reduction
BC.Se.inferCNV <- FindVariableFeatures(BC.Se.inferCNV, selection.method = "vst",nfeatures = 6000)
BC.Se.inferCNV <- ScaleData(BC.Se.inferCNV, assay = "RNA", verbose = FALSE)
BC.Se.inferCNV <- RunPCA(BC.Se.inferCNV, verbose = FALSE)
BC.Se.inferCNV <- RunUMAP(BC.Se.inferCNV, dims = 1:30, verbose = FALSE)

DimPlot(BC.Se.inferCNV, group.by = "celltype_major")

```

```{r reading in the proteomics data}
protein_normalized <- read_csv("/mnt/data1/elyas/Single-cell/Pseudo_bulk-proteomics/protein_normalized.csv")
protein_normalized <- as.data.frame(protein_normalized[!duplicated(protein_normalized$Gene_Symbol),])

protein_normalized <- protein_normalized[complete.cases(protein_normalized$Gene_Symbol), ]
rownames(protein_normalized) <- protein_normalized$Gene_Symbol
protein.table <- protein_normalized[,-c(1:6, 20:22)]
protein.table <- protein.table[,-c(14,17,24,26)]
protein.table[is.na(protein.table)] = 0

protein_seurat <- CreateSeuratObject(protein.table, min.cells = 3, min.genes = 200, project = "Protein" )
protein_seurat@assays$RNA@data <- protein_seurat@assays$RNA@counts
protein_seurat <- FindVariableFeatures(protein_seurat, nfeatures = 6000)
protein_seurat <- ScaleData(protein_seurat, verbose = FALSE)
protein_seurat <- RunPCA(protein_seurat, verbose = FALSE,npcs = 15)
protein_seurat <- RunUMAP(protein_seurat, dims = 1:15, verbose = FALSE,n.neighbors =24 )
protein_seurat$subtype <- c(rep("HER2",6), rep("ER",7), rep("TNBC",11))
protein_seurat <- SetIdent(protein_seurat, value = protein_seurat$subtype)

```

```{r}
save(BC.Se.inferCNV, BC.Se.mal,CD4,protein_seurat,file = "../Github/Step1.Data_processing.Rdata")

```


